book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,195 @@
+| Sapling..Blossom                       | 80%      | 20%            | 0%              | No                |
+| Blossom..Heartwood                     | 80%      | 20%            | 0%              | No                |
+| Heartwood..Canopy                      | 80%      | 20%            | 0%              | Yes               |
+| Canopy..Second Halving                 | 80%      | 0%             | 20%             | Yes               |
+| ~Second Halving..~                     | ~100%~   | ~0%~           | ~0%~            | ~Yes~             |
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Given the module structure proposed above all the block subsidy calculations from the protocol must be implemented. The final goal is to do checks for each incoming block and make sure they pass the consensus rules.
+
+To do the calculations and checks the following constants and functions need to be introduced:
+
+## Constants
+
+- `SLOW_START_INTERVAL`
+- `SLOW_START_SHIFT`
+- `MAX_BLOCK_SUBSIDY`
+- `PRE_BLOSSOM_POW_TARGET_SPACING`
+- `POST_BLOSSOM_POW_TARGET_SPACING`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO`
+- `PRE_BLOSSOM_HALVING_INTERVAL`
+- `POST_BLOSSOM_HALVING_INTERVAL`
+- `FOUNDERS_FRACTION_DIVISOR`
+- `FOUNDER_ADDRESS_CHANGE_INTERVAL`
+- `FOUNDERS_REWARD_ADDRESSES_MAINNET`
+- `FOUNDERS_REWARD_ADDRESSES_TESTNET`
+
+### Funding streams parameter constants
+
+The design suggests to implement the parameters needed for funding streams as:
+
+```
+/// The funding stream receiver categories
+pub enum FundingStreamsReceivers {
+    /// Electric Coin Company
+    ECC,
+    /// ZCash Foundation
+    ZF,
+    /// Major Grants
+    MG,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_RECEIVERS_NUMERATORS: &[(i32, FundingStreamsReceivers)] = &[
+    (7, FundingStreamsReceivers::ECC),
+    (5, FundingStreamsReceivers::ZF),
+    (8, FundingStreamsReceivers::MG),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDINGSTREAM_DENOMINATOR: i32 = 100;
+
+/// Height intervals for funding streams on each network.
+pub enum FundingStreamsHeightIntervals {
+    /// For the Mainnet
+    Mainnet,
+    /// For the Testnet
+    Testnet,
+}
+
+/// Start and end Heights for funding streams 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_HEIGHT_INTERVALS: &[(Height, Height, FundingStreamsHeightIntervals)] = &[
+    (Height(1_046_400), Height(2_726_400), FundingStreamsHeightIntervals::Mainnet),
+    (Height(1_028_500), Height(2_796_000), FundingStreamsHeightIntervals::Testnet),
+];
+```
+
```

Comment:
This design is ok, but I think it will be easier to use if we make new types for:
- [x] `FundingStreamReceiver`
- [ ] `FundingStreamRange`
- [ ] and return the consensus parameters using functions implemented on those types

For an example, see `NetworkUpgrade` in:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/parameters/network_upgrade.rs

When you're making a data array:
- [ ] try to put the unique values first  (FundingStreamReceiver, Network)

That makes lookups using a `BTreeMap` easier. And it makes the tests easier.